packages/@aws-cdk/aws-eks/lib/cluster.ts

```diff
@@ -981,6 +997,12 @@ abstract class ClusterBase extends Resource implements ICluster {
     if (autoScalingGroup.spotPrice && addSpotInterruptHandler) {
       this.addSpotInterruptHandler();
     }
+
+    if (this instanceof Cluster) {
```

Comment:
I've seen this pattern before in our code base. So just out of curiosity: why not place this logic inside `Cluster` itself?

Reply:
The implementation of `connectAutoScalingGroupCapacity` is inside `ClusterBase`, which can also be an imported cluster...I guess we can create an implementation inside `Cluster` which overrides the base one and adds this, but doesn't really make much of a difference. 

Or did you mean something else?

Reply:
I mean in `Cluster`:

```ts
public addServiceAccount(id: string, options: ServiceAccountOptions = {}): ServiceAccount {
  super.addServiceAccount(id, options);
  this.albController?.node.addDependency(autoScalingGroup);
}
```

Reply:
Yeap in this case we can definitely do that. In general sometimes it might not be possible in case the interjection needs to happen somewhere inside the original implementation. 

In any case, i'll leave it as is for now...but good point.